charts/redpanda/templates/_configmap.tpl

```diff
@@ -130,7 +130,7 @@ redpanda.yaml: |
 {{- end }}
 {{- with dig "node" dict .Values.config }}
   {{- range $key, $element := .}}
-    {{- if and (or (eq (typeOf $element) "bool") $element) (and (eq $key "crash_loop_limit") (include "redpanda-atleast-23-1-1" $root | fromJson).bool) }}
+    {{- if and (or (eq (typeOf $element) "bool") $element) (or (and (eq $key "crash_loop_limit") (include "redpanda-atleast-23-1-1" $root | fromJson).bool) (ne $key "crash_loop_limit")) }}
     {{ $key }}: {{ $element | toYaml }}
     {{- end }}
   {{- end }}
```

Comment:
Another way of doing this that doesn't build those long complex (and error prone) if statements:

```suggestion
  {{- range $key, $element := .}}
    {{- $line := dict $key (toYaml $element) }}
    {{- if and (eq $key "crash_loop_limit") (include "redpanda-atleast-23-1-1" $root | fromJson).bool) }}
      {{- $line = dict }}
    {{- end }}
    {{- if not (or (eq (typeOf $element) "bool") $element) }}
      {{- $line = dict }}
    {{- with $line}
      {{  . | nindent 4 }}
    {{- end }}
  {{- end }}
  ```

Reply:
added a commit with this change